encoding/src/commonMain/kotlin/BitSet.kt

```diff
@@ -27,6 +30,12 @@ public data class IntBitSet(private var buffer: Int) : BitSet<Int> {
     }
 
     override fun asPrimitive(): Int = buffer
+
+    override fun extract(offset: Int, count: Int): Int {
+        require(offset in 0 until 32)
+        require(count in 0 until 32 - offset + 1)
```

Comment:
nit: Can we give these failure messages (that includes the value that was provided), please?